modules/host/network.go

```diff
@@ -49,6 +49,75 @@ func (h *Host) threadedUpdateHostname(closeChan chan struct{}) {
 	}
 }
 
+// threadedTrackWorkingState periodically checks if the host is working,
+// where working is defined as having received 3 settings calls in the past 15
+// minutes.
+func (h *Host) threadedTrackWorkingState(closeChan chan struct{}) {
+	defer close(closeChan)
+
+	for {
+		prevSettingsCalls := atomic.LoadUint64(&h.atomicSettingsCalls)
+
+		select {
+		case <-h.tg.StopChan():
+			return
+		case <-time.After(workingStateFrequency):
+		}
+
+		settingsCalls := atomic.LoadUint64(&h.atomicSettingsCalls)
```

Comment:
can probably drop the newlines buffering the soft-sleep pattern. The whole idea of get calls, sleep, get calls again is a single thought.